README
NAME

    Dist::Zilla::Plugin::Authority - Add the $AUTHORITY variable and
    metadata to your distribution

VERSION

      This document describes v1.008 of Dist::Zilla::Plugin::Authority - released October 27, 2014 as part of Dist-Zilla-Plugin-Authority.

DESCRIPTION

    This plugin adds the authority data to your distribution. It adds the
    data to your modules and metadata. Normally it looks for the PAUSE
    author id in your Dist::Zilla configuration. If you want to override
    it, please use the 'authority' attribute.

            # In your dist.ini:
            [Authority]

    This code will be added to any package declarations in your perl files:

            our $AUTHORITY = 'cpan:APOCAL';

    Your metadata ( META.yml or META.json ) will have an entry looking like
    this:

            x_authority => 'cpan:APOCAL'

ATTRIBUTES

 authority

    The authority you want to use. It should be something like cpan:APOCAL.

    Defaults to the username set in the %PAUSE stash in the global
    config.ini or dist.ini ( Dist::Zilla v4 addition! )

    If you prefer to not put it in config/dist.ini you can put it in
    "~/.pause" just like Dist::Zilla did before v4.

 do_metadata

    A boolean value to control if the authority should be added to the
    metadata.

    Defaults to true.

 do_munging

    A boolean value to control if the $AUTHORITY variable should be added
    to the modules.

    Defaults to true.

 locate_comment

    A boolean value to control if the $AUTHORITY variable should be added
    where a # AUTHORITY comment is found. If this is set then an
    appropriate comment is found, and our $AUTHORITY = 'cpan:PAUSEID'; is
    inserted preceding the comment on the same line.

    This basically implements what OurPkgVersion does for PkgVersion.

    Defaults to false.

    NOTE: If you use this method, then we will not use the pkg style of
    declaration! That way, we keep the line numbering consistent.

 authority_style

    A value to control the type of the $AUTHORITY declaration. There are
    two styles: 'pkg' or 'our'. In the past this module defaulted to the
    'pkg' style but due to various issues 'our' is now the default. Here's
    what both styles would look like in the resulting code:

            # pkg
            BEGIN {
                    $Dist::Zilla::Plugin::Authority::AUTHORITY = 'cpan:APOCAL';
            }
    
            # our
            our $AUTHORITY = 'cpan:APOCAL';

SEE ALSO

    Please see those modules/websites for more information related to this
    module.

      * Dist::Zilla

      * http://www.perlmonks.org/?node_id=694377

      * http://perlcabal.org/syn/S11.html#Versioning

SUPPORT

 Perldoc

    You can find documentation for this module with the perldoc command.

      perldoc Dist::Zilla::Plugin::Authority

 Websites

    The following websites have more information about this module, and may
    be of help to you. As always, in addition to those websites please use
    your favorite search engine to discover more resources.

      * MetaCPAN

      A modern, open-source CPAN search engine, useful to view POD in HTML
      format.

      http://metacpan.org/release/Dist-Zilla-Plugin-Authority

      * Search CPAN

      The default CPAN search engine, useful to view POD in HTML format.

      http://search.cpan.org/dist/Dist-Zilla-Plugin-Authority

      * RT: CPAN's Bug Tracker

      The RT ( Request Tracker ) website is the default bug/issue tracking
      system for CPAN.

      http://rt.cpan.org/NoAuth/Bugs.html?Dist=Dist-Zilla-Plugin-Authority

      * AnnoCPAN

      The AnnoCPAN is a website that allows community annotations of Perl
      module documentation.

      http://annocpan.org/dist/Dist-Zilla-Plugin-Authority

      * CPAN Ratings

      The CPAN Ratings is a website that allows community ratings and
      reviews of Perl modules.

      http://cpanratings.perl.org/d/Dist-Zilla-Plugin-Authority

      * CPAN Forum

      The CPAN Forum is a web forum for discussing Perl modules.

      http://cpanforum.com/dist/Dist-Zilla-Plugin-Authority

      * CPANTS

      The CPANTS is a website that analyzes the Kwalitee ( code metrics )
      of a distribution.

      http://cpants.cpanauthors.org/dist/overview/Dist-Zilla-Plugin-Authori
      ty

      * CPAN Testers

      The CPAN Testers is a network of smokers who run automated tests on
      uploaded CPAN distributions.

      http://www.cpantesters.org/distro/D/Dist-Zilla-Plugin-Authority

      * CPAN Testers Matrix

      The CPAN Testers Matrix is a website that provides a visual overview
      of the test results for a distribution on various Perls/platforms.

      http://matrix.cpantesters.org/?dist=Dist-Zilla-Plugin-Authority

      * CPAN Testers Dependencies

      The CPAN Testers Dependencies is a website that shows a chart of the
      test results of all dependencies for a distribution.

      http://deps.cpantesters.org/?module=Dist::Zilla::Plugin::Authority

 Email

    You can email the author of this module at APOCAL at cpan.org asking
    for help with any problems you have.

 Internet Relay Chat

    You can get live help by using IRC ( Internet Relay Chat ). If you
    don't know what IRC is, please read this excellent guide:
    http://en.wikipedia.org/wiki/Internet_Relay_Chat. Please be courteous
    and patient when talking to us, as we might be busy or sleeping! You
    can join those networks/channels and get help:

      * irc.perl.org

      You can connect to the server at 'irc.perl.org' and join this
      channel: #perl-help then talk to this person for help: Apocalypse.

      * irc.freenode.net

      You can connect to the server at 'irc.freenode.net' and join this
      channel: #perl then talk to this person for help: Apocal.

      * irc.efnet.org

      You can connect to the server at 'irc.efnet.org' and join this
      channel: #perl then talk to this person for help: Ap0cal.

 Bugs / Feature Requests

    Please report any bugs or feature requests by email to
    bug-dist-zilla-plugin-authority at rt.cpan.org, or through the web
    interface at
    http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Dist-Zilla-Plugin-Author
    ity. You will be automatically notified of any progress on the request
    by the system.

 Source Code

    The code is open to the world, and available for you to hack on. Please
    feel free to browse it and play with it, or whatever. If you want to
    contribute patches, please send me a diff or prod me to pull from your
    repository :)

    https://github.com/apocalypse/perl-dist-zilla-plugin-authority

      git clone git://github.com/apocalypse/perl-dist-zilla-plugin-authority.git

AUTHOR

    Apocalypse <APOCAL@cpan.org>

 CONTRIBUTORS

      * Dave Rolsky <autarch@urth.org>

      * Kent Fredric <kentnl@cpan.org>

      * Nigel Metheringham <nigel.metheringham@dev.intechnology.co.uk>

      * Randy Stauner <randy@magnificent-tears.com>

ACKNOWLEDGEMENTS

    This module is basically a rip-off of RJBS' excellent
    Dist::Zilla::Plugin::PkgVersion, thanks!

    Props goes out to FLORA for prodding me to improve this module!

COPYRIGHT AND LICENSE

    This software is copyright (c) 2014 by Apocalypse.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.

    The full text of the license can be found in the LICENSE file included
    with this distribution.

DISCLAIMER OF WARRANTY

    THERE IS NO WARRANTY FOR THE PROGRAM, TO THE EXTENT PERMITTED BY
    APPLICABLE LAW. EXCEPT WHEN OTHERWISE STATED IN WRITING THE COPYRIGHT
    HOLDERS AND/OR OTHER PARTIES PROVIDE THE PROGRAM "AS IS" WITHOUT
    WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING, BUT NOT
    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
    PARTICULAR PURPOSE. THE ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE
    OF THE PROGRAM IS WITH YOU. SHOULD THE PROGRAM PROVE DEFECTIVE, YOU
    ASSUME THE COST OF ALL NECESSARY SERVICING, REPAIR OR CORRECTION.

    IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
    WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MODIFIES AND/OR
    CONVEYS THE PROGRAM AS PERMITTED ABOVE, BE LIABLE TO YOU FOR DAMAGES,
    INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES
    ARISING OUT OF THE USE OR INABILITY TO USE THE PROGRAM (INCLUDING BUT
    NOT LIMITED TO LOSS OF DATA OR DATA BEING RENDERED INACCURATE OR LOSSES
    SUSTAINED BY YOU OR THIRD PARTIES OR A FAILURE OF THE PROGRAM TO
    OPERATE WITH ANY OTHER PROGRAMS), EVEN IF SUCH HOLDER OR OTHER PARTY
    HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.



Changes
Revision history for Dist::Zilla::Plugin::Authority

1.008	2014-10-27 21:33:09 UTC

	Made sure that we cannot be loaded with different authorities via RT#86710 ( thanks ETHER! )

1.007	2014-10-27 20:42:01 UTC

	Rewrote how we assign the $AUTHORITY variable to use 'our' instead of the crufty pkg declaration format via gh-5 and gh-6 and RT#86611 ( thanks KENTNL + ETHER )

1.006	2012-01-03 04:14:27 UTC

	Use the new PPI role in Dist::Zilla 4.300001 to make this plugin faster ( thanks DROLSKY! )

1.005	2011-04-14 17:05:17 UTC

	Added locate_comment which finds a # AUTHORITY comment and puts authority there ( thanks NIGELM! )

1.004	2011-02-21 17:55:27 UTC

	Tweak the authority regex to be a bit more liberal ( thanks RWSTAUNER! )

1.003	2011-02-04 07:24:46 UTC

	Tweaked the ABSTRACT section ( thanks FLORA + GETTY )
	Add ability to read from ~/.pause file for the authority ( thanks WOLVERIAN )

1.002	2010-12-14 01:32:18 UTC

	Added the "do_munging" attr to control the variable declarations in packages ( thanks FLORA )

1.001	2010-12-08 02:36:12 UTC

	Updated to require Dist::Zilla v4 ( so we can access the %PAUSE stash )
	The "authority" attr is not required anymore ( defaults to the %PAUSE username )
	Backported some code from Dist::Zilla::Plugin::PkgVersion v4 ( as usual, thanks RJBS )
	Add $AUTHORITY to script files too ( if they have a package declaration )
	Make do_metadata default to true ( thanks FLORA )

1.000	2010-05-30 17:43:51 UTC

	added the patch from Dave Rolsky for "monkeypatch package declaration" fix
	Converted to Dist::Zilla for the release process!

0.02	0000-00-00 00:00:00 UTC

	added the patch from Dist::Zilla::Plugin::PkgVer v2.101310 for multi-package declarations

0.01	0000-00-00 00:00:00 UTC

	initial release


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.021.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.28"
  },
  "configure_requires" => {
    "Module::Build" => "0.28"
  },
  "dist_abstract" => "Add the \$AUTHORITY variable and metadata to your distribution",
  "dist_author" => [
    "Apocalypse <APOCAL\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-Authority",
  "dist_version" => "1.008",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Plugin::Authority",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Dist::Zilla::Role::FileFinderUser" => "4.102345",
    "Dist::Zilla::Role::FileMunger" => "4.102345",
    "Dist::Zilla::Role::MetaProvider" => "4.102345",
    "Dist::Zilla::Role::PPI" => "4.300001",
    "Dist::Zilla::Util" => 0,
    "File::HomeDir" => 0,
    "File::Spec" => 0,
    "Moose" => "1.03",
    "Moose::Util::TypeConstraints" => "1.01",
    "PPI" => "1.206",
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.88",
    "perl" => "5.006"
  }
);


my %fallback_build_requires = (
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Module::Build" => "0.28",
  "Test::More" => "0.88",
  "perl" => "5.006"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.021.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Add the \$AUTHORITY variable and metadata to your distribution",
  "AUTHOR" => "Apocalypse <APOCAL\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Module::Build" => "0.28"
  },
  "CONFIGURE_REQUIRES" => {
    "Module::Build" => "0.28"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Authority",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::Authority",
  "PREREQ_PM" => {
    "Dist::Zilla::Role::FileFinderUser" => "4.102345",
    "Dist::Zilla::Role::FileMunger" => "4.102345",
    "Dist::Zilla::Role::MetaProvider" => "4.102345",
    "Dist::Zilla::Role::PPI" => "4.300001",
    "Dist::Zilla::Util" => 0,
    "File::HomeDir" => 0,
    "File::Spec" => 0,
    "Moose" => "1.03",
    "Moose::Util::TypeConstraints" => "1.01",
    "PPI" => "1.206",
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.88"
  },
  "VERSION" => "1.008",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::Role::FileFinderUser" => "4.102345",
  "Dist::Zilla::Role::FileMunger" => "4.102345",
  "Dist::Zilla::Role::MetaProvider" => "4.102345",
  "Dist::Zilla::Role::PPI" => "4.300001",
  "Dist::Zilla::Util" => 0,
  "File::HomeDir" => 0,
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Module::Build" => "0.28",
  "Moose" => "1.03",
  "Moose::Util::TypeConstraints" => "1.01",
  "PPI" => "1.206",
  "Test::More" => "0.88",
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



